app/src/main/java/zapsolutions/zap/util/ExchangeRateUtil.java

```diff
@@ -0,0 +1,308 @@
+            try {
+                JSONObject ReceivedCurrency = response.getJSONObject(fiatCode);
+                JSONObject currentCurrency = new JSONObject();
+                currentCurrency.put(RATE, ReceivedCurrency.getDouble("15m") / 1e8);
+                currentCurrency.put(SYMBOL, ReceivedCurrency.getString("symbol"));
+                currentCurrency.put(TIMESTAMP, System.currentTimeMillis() / 1000);
+                formattedRates.put(fiatCode, currentCurrency);
+            } catch (JSONException e) {
+                ZapLog.debug(LOG_TAG, "Unable to read exchange rate from blockchain.info response.");
+            }
+        }
+
+        // Switch order as blockchain.info has USD first. We want to have it alphabetically.
+        try {
+            JSONObject tempCurrency = formattedRates.getJSONObject("USD");
+            formattedRates.remove("USD");
+            formattedRates.put("USD", tempCurrency);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        return formattedRates;
+    }
+
+    /**
+     * This function parses a coinbase exchange rate response.
+     * All the response parser functions return a similar formatted JSON Object
+     * {"USD":{"rate":0.1231, "timestamp":...},"EUR":{...}}
+     *
+     * @param response a JSON response that comes from Coinbases API
+     * @return
+     */
+    public JSONObject parseCoinbaseResponse(JSONObject response) {
+
+        JSONObject responseRates = null;
+        try {
+            responseRates = response.getJSONObject("data").getJSONObject("rates");
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        JSONObject formattedRates = new JSONObject();
+        if (responseRates != null) {
+            Iterator<String> iter = responseRates.keys();
+            while (iter.hasNext()) {
+                String rateCode = iter.next();
+
+                try {
+                    JSONObject currentCurrency = new JSONObject();
+                    currentCurrency.put(RATE, responseRates.getDouble(rateCode) / 1e8);
+                    currentCurrency.put(TIMESTAMP, System.currentTimeMillis() / 1000);
+                    formattedRates.put(rateCode, currentCurrency);
+                } catch (JSONException e) {
+                    ZapLog.debug(LOG_TAG, "Unable to read exchange rate from coinbase response.");
+                }
+            }
+        } else {
+            return null;
+        }
+
+        return formattedRates;
+    }
+
+    /**
+     * Some responses include exchange rates to other crypto currencies.
+     * This function removes them by checking if the currency code is in our fiat currency list.
+     */
+    private JSONObject removeNonFiat(JSONObject rates) {
+        Iterator<String> iter = rates.keys();
+        while (iter.hasNext()) {
+            String rateCode = iter.next();
+            if (AppUtil.getInstance(mContext).getCurrencyNameFromCurrencyCode(rateCode) == null) {
```

Comment:
Hmm, not sure but this loads every iteration the whole json list from resources. Maybe a little cost expensive.
Would be nice if we can load the list once and then use it.